manim/renderer/cairo_renderer.py

```diff
@@ -71,12 +38,55 @@ def init_scene(self, scene):
             scene.__class__.__name__,
         )
 
-    @pass_scene_reference
-    @handle_caching_play
-    @handle_play_like_call
     def play(self, scene, *args, **kwargs):
-        if scene.compile_animation_data(*args, **kwargs):
+        # Reset skip_animations to the original state.
+        # Needed when rendering only some animations, and skipping others.
+        self.skip_animations = self._original_skipping_status
+        self.update_skipping_status()
+
+        scene.compile_animation_data(*args, **kwargs)
+
+        # If skip_animations is already True, we can skip all the caching process.
+        if not config["disable_caching"] and not self.skip_animations:
+            hash_current_animation = get_hash_from_play_call(
+                scene, self.camera, scene.animations, scene.mobjects
+            )
+            if self.file_writer.is_already_cached(hash_current_animation):
+                logger.info(
+                    f"Animation {self.num_plays} : Using cached data (hash : %(hash_current_animation)s)",
+                    {"hash_current_animation": hash_current_animation},
+                )
+                self.skip_animations = True
+        else:
+            hash_current_animation = f"uncached_{self.num_plays:05}"
+
+        if self.skip_animations:
+            logger.debug(f"Skipping animation {self.num_plays}")
+            hash_current_animation = None
+
+        # adding None as a partial movie file will make file_writer ignore the latter.
+        self.file_writer.add_partial_movie_file(hash_current_animation)
+        self.animations_hashes.append(hash_current_animation)
+        logger.debug(
+            "List of the first few animation hashes of the scene: %(h)s",
+            {"h": str(self.animations_hashes[:5])},
+        )
+
+        # Save a static image, to avoid rendering non moving objects.
+        self.static_image = self.save_static_frame_data(scene, scene.static_mobjects)
```

Comment:
Altogether, the adjective "static" isn't helpful here since an image is already understood to be non-moving media. A better adjective might be "stationary", or "frozen" so it's not confused with static (as in `@staticmethod`) but it's not the image that is stationary/frozen, it's the Animation/the Mobjects in the Animation.
```suggestion
        # Save a image, to avoid rendering non moving objects.
        self.image = self.save_frame_data(scene, scene.static_mobjects)
```

Reply:
Just for the record, this code is not mine (I moved this piece from elsewhere to here), the term `static_image` is braodly used to say that an image is static. 

I believe this is not the right PR toc hange this, because it was not the main goal, but it's defintely an issue.

Reply:
It's just an issue of better variable names, but if it helps move things along quicker then I'm fine with doing this in a future PR.